src/contrib/transports/Akka.Remote.Transport.Streams.Tcp/Class1.cs

```diff
@@ -0,0 +1,313 @@
+                                    new Disassociated(DisassociateInfo.Unknown));
+                                return IO.ByteString.Empty;
+                            }), _mat);
+                    queuePromise.SetResult(outQueue);
+                    listener.Notify(new InboundAssociation(handle));
+                });
+                return NotUsed.Instance;
+            }).ToMaterialized(Sink.Ignore<NotUsed>(),Keep.Left).Run(_mat);
+            return (_addr, AssociationListenerPromise);
+        }
+
+        
+        protected async Task<IPEndPoint> DnsToIPEndpoint(DnsEndPoint dns)
+        {
+            IPEndPoint endpoint;
+
+            var addressFamily = Settings.DnsUseIpv6 ? AddressFamily.InterNetworkV6 : AddressFamily.InterNetwork;
+            endpoint = await DnsHelpers.ResolveNameAsync(dns, addressFamily).ConfigureAwait(false);
+
+            return endpoint;
+        }
+        public override bool IsResponsibleFor(Address remote)
+        {
+            return true;
+        }
+
+        public override async Task<AssociationHandle> Associate(Address remoteAddress)
+        {
+            
+            var c =base.System.TcpStream().OutgoingConnection(
+                DnsHelpers.AddressToSocketAddress(remoteAddress),
+                _outAddr,
+                ImmutableList.Create<Inet.SocketOption>(
+                    new Inet.SO.ReceiveBufferSize(
+                        Settings.ReceiveBufferSize ?? 256 * 1024),
+                    new Inet.SO.SendBufferSize(
+                        Settings.SendBufferSize ?? 256 * 1024)));
+            
+            
+            var queuePromise =
+                    new TaskCompletionSource<
+                        ISourceQueueWithComplete<IO.ByteString>>();
+            var handle = new StreamingTcpAssociationHandle(_addr,remoteAddress, queuePromise);
+            handle.ReadHandlerSource.Task.ContinueWith(s =>
+            {   
+                var listener = s.Result;
+                handle.RegisterListener(listener);
+            }, TaskContinuationOptions.ExecuteSynchronously);
+            var run = c.JoinMaterialized(buildFlow(handle).Recover(
+                    ex =>
+                    {
+                        handle.Notify(
+                            new Disassociated(DisassociateInfo.Unknown));
+                        return IO.ByteString.Empty;
+                    }),(oc,sq)=>(oc,sq)).Run(_mat);
+            queuePromise.SetResult(run.sq);
+            return handle;
+        }
+
+        
+        private Source<IO.ByteString, ISourceQueueWithComplete<IO.ByteString>> sendFlow()
+        {
+            //TODO: Improve batching voodoo in pipeline.
+            return Source
+                .Queue<IO.ByteString>(512, OverflowStrategy.DropNew)
+                .Via(
+                    Framing.SimpleFramingProtocolEncoder(Settings.MaxFrameSize))
+                .GroupedWithin(256, TimeSpan.FromMilliseconds(40))
```

Comment:
Nice, a much more concise way of doing the "batching" than what our current transport does. We'll still need a flag to disable the batching stage via config for connections with low traffic (i.e. a system like Lighthouse)

Reply:
Not gonna lie, I want to FRAME that block of code. While I know there's room for improvement it just feels so elegant.